lua/neogit/popups/init.lua

```diff
@@ -0,0 +1,109 @@
+local M = {}
+
+---@param name string
+---@param f nil|fun(create: fun(...)): any
+--- Creates a curried function which will open the popup with the given name when called
+function M.open(name, f)
+  f = f or function(f)
+    f()
+  end
+
+  return function()
+    local ok, value = pcall(require, "neogit.popups." .. name)
+    if ok then
+      assert(value, "popup is not nil")
+      assert(value.create, "popup has create function")
+
+      f(value.create)
+    else
+      local notification = require("neogit.lib.notification")
+      notification.create(string.format("No such popup: %q", name), vim.log.levels.ERROR)
+    end
+  end
+end
+
+--- Returns an array useful for creating mappings for the available popups
+---@return table<string, Mapping>
+function M.mappings_table()
+  local config = require("neogit.config")
+  local async = require("plenary.async")
+  return {
+    {
+      "HelpPopup",
+      "Help",
+      M.open("help", function(f)
+        f {
+          use_magit_keybindings = config.values.use_magit_keybindings,
```

Comment:
I'm a little torn on this pattern of reaching into the status buffer for stuff. On the one hand, its a little cleaner than needing to pass the line into the constructor, but on the other hand it makes this function dependent on external state instead of passing that state in as an argument.

Not saying to change this, just voicing my thoughts

Reply:
Yes, I am as well.

I usually go about coding with a domain driven approach. Some code lives in a certain domain/space, and can enter a smaller contained domain. Code usually becomes messier when a code enters a larger domain than which is was called from. Think of a graph which has to remain acyclic, like a tree.

The status buffer would need a bit of splitting like you did for the repository.

Reply:
Perhaps splitting it as well into an `init/view` and `actions` and then using actions in other places.

Reply:
Splitting the status buffer up is a longer-term goal of mine, but it clearly will take some doing. It also pre-dates the buffer lib, so doesn't really take advantage of the library that all the popups/buffers do. _Eventually_ I'd like the status buffer to not have a special place like it currently does, and just be one more view (albeit the "main" view), but given the size/complexity of it, thats not a simple undertaking. So, small bits here and there.

It may be a good middle-step to use the `init/view`/`actions` pattern - I'm not against that.